### PR TITLE
feat: added new UI components and reorganized routing logic (#300)

### LINKED ISSUE #300: [SystemContract DApp] new UI for the Overview
### Problem

Overview UI needs updating

### Solution

new UI for the Overview,

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**: This PR is mainly for UI and routing changes

- reworked protected routes logic
- reworked layout logic
- added protected pages for smart contracts
- added new LeftSideBar UI component
- added new Overview UI component

**UI updates**:
route: /hedera/overview
image

route: /hedera/[system_contract]
image

image

**Related issue(s)**:

Fixes #300 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: system-contract-dapp-playground/src/app/hedera/hrc-719/page.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const HRC719 = () => {
+  return (
+    <div className="text-white w-full flex justify-center items-center text-[3rem]">HRC719</div>
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/app/hedera/hrc-719/page.tsx:23]:
```suggestion
    <div className="text-white w-full flex justify-center items-center text-[3rem]">Token Associate Facade (HRC719)</div>
```

File: system-contract-dapp-playground/src/app/hedera/hts-hip-206/page.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const HTS206 = () => {
+  return (
+    <div className="text-white w-full flex justify-center items-center text-[3rem]">HTS206</div>
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/app/hedera/hts-hip-206/page.tsx:23]:
```suggestion
    <div className="text-white w-full flex justify-center items-center text-[3rem]">Hedera Token Sservice (HTS) System Contract (HIP 206)</div>
```

File: system-contract-dapp-playground/src/app/hedera/prng-hip-351/page.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const HIP351 = () => {
+  return (
+    <div className="text-white w-full flex justify-center items-center text-[3rem]">HIP351</div>
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/app/hedera/prng-hip-351/page.tsx:23]:
```suggestion
    <div className="text-white w-full flex justify-center items-center text-[3rem]">Pseudo Random Number Generator (PRNG System Contract) (HIP 351)</div>
```

File: system-contract-dapp-playground/src/utils/constants.ts [hunk 4]
```diff
@@ -111,14 +111,48 @@ export const HEDERA_NETWORKS = {
...
+/**
+ * @notice information for LeftSideBar items
+ */
+export const LEFT_SIDE_BAR_ITEMS = [
+  {
+    name: 'Overview',
+    path: PROTECTED_ROUTES[0],
+  },
+  {
+    name: 'HTS system contract wrapper (HIP-206)',
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/utils/constants.ts:135]: Better yet make the page labels match these.

**Review by Nana-EC — COMMENTED** [review]
Looking good.
Update page headings

File: system-contract-dapp-playground/src/app/hedera/exchange-rate-hip-475/page.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const HIP475 = () => {
+  return (
+    <div className="text-white w-full flex justify-center items-center text-[3rem]">HIP475</div>
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply system-contract-dapp-playground/src/app/hedera/exchange-rate-hip-475/page.tsx:23]: Oh ah yeah these are temporary values which will be replaced by real component in the future PRs. These are just to mark which route is which temporarily. Bu thanks for pointing it out!

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**